File: api/src/main/java/net/blueshell/api/dto/ContributionDTO.java
package net.blueshell.api.dto;

import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.NotNull;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.sql.Timestamp;

@Data
@EqualsAndHashCode(callSuper = false)
@Schema(name = "Contribution")
public class ContributionDTO extends BaseDTO {
    private Long id;

    @NotNull
    private Long userId;

    @NotNull
    private Long contributionPeriodId;

    @NotNull
    private Boolean paid;

    private Timestamp remindedAt;
}

File: api/src/main/kotlin/net/blueshell/api/dto/contribution/ContributionDTO.kt
package net.blueshell.api.dto.contribution

import io.swagger.v3.oas.annotations.media.Schema
import jakarta.validation.constraints.NotNull
import net.blueshell.api.base.BaseDTO
import java.sql.Timestamp

@Schema(name = "Contribution")
data class ContributionDTO(
    @field:NotNull
    var userId: Long? = null,

    @field:NotNull
    var contributionPeriodId: Long? = null,

    var remindedAt: Timestamp? = null
) : BaseDTO()
